test/e2e/examples_test.go

```diff
@@ -112,12 +112,50 @@ func TestCppSimpleGameServerReady(t *testing.T) {
 				ContainerPort: 7654,
 				Protocol:      corev1.ProtocolUDP,
 			}},
+			Health: agonesv1.Health{
+				InitialDelaySeconds: 300,
```

Comment:
Okay, time for a new theory! So I don't think it's the `InitialDelaySeconds` -- the failure is specifically happening on the Autopilot cluster (but let's leave this alone, we can always remove this later).

I think it's a Kuerbernetes `resource` issue!

* https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/

Here is a yaml example (we can happily just set requests, Autopilot will copy the limit over)
https://github.com/googleforgames/agones/blob/9e4400b213e970db7021ac7f7c930e63fb7b8107/examples/simple-game-server/fleet.yaml#L31-L34

Lets add that to the PodSpec -> container -- I think that will resolve this issue.